app/scripts/modules/core/diffs/viewChangesLink.component.ts

```diff
@@ -0,0 +1,72 @@
+import {IComponentController, IComponentOptions, module} from 'angular';
+import {IModalInstanceService, IModalService} from 'angular-ui-bootstrap';
+
+import {ICommit} from './commitHistory.component';
+import {COMMIT_HISTORY_COMPONENT} from './commitHistory.component';
+import {JAR_DIFF_COMPONENT, IJarDiff} from './jarDiff.component';
+
+interface INameItem {
+  name: string;
+}
+
+class ViewChangesModalController {
+
+  static get $inject(): string[] {
+    return ['$uibModalInstance', 'commits', 'jarDiffs', 'nameItem'];
+  }
+
+  constructor(private $uibModalInstance: IModalInstanceService,
+              public commits: ICommit[],
+              public jarDiffs: IJarDiff,
+              public nameItem: INameItem) {}
+
+  public close(): void {
+    this.$uibModalInstance.close();
+  }
+}
+
+class ViewChangesLinkController implements IComponentController {
+
+  public commits: ICommit[];
+  public jarDiffs: IJarDiff;
+  public linkText: string;
+  public nameItem: INameItem;
+
+  static get $inject(): string[] {
+    return ['$uibModal'];
+  }
+
+  constructor(private $uibModal: IModalService) {}
+
+  public $onInit(): void {
+    this.linkText = this.linkText || 'View Changes';
```

Comment:
Sorry, to clarify: if you change line 32 to:
```
public linkText = 'View Changes';
```
do you still need the `$onInit` method, i.e. does Angular replace the value if it's passed in to the component?